File: internal/projects/domain/project/project.go
package project

import "social-media/internal/projects/domain/user"

type Project struct {
	id          int
	name        string
	description string
	stack       string
	user        *user.User
}

func New(id int, name, description, stack string, userID int) *Project {
	return &Project{
		id:          id,
		name:        name,
		description: description,
		stack:       stack,
		user: user.New(userID, ""),
	}
}

func (p *Project) ID() int {
	return p.id
}

func (p *Project) Name() string {
	return p.name
}

func (p *Project) Description() string {
	return p.description
}

func (p *Project) Stack() string {
	return p.stack
}

func (p *Project) UserID() int {
	return p.user.ID()
}

func (p *Project) UserLogin() string {
	return p.user.Login()
}

func (p *Project) SetUserLogin(login string) {
	p.user.SetLogin(login)
}

File: client/src/app/objects/project.ts
import { GetProjectResponse } from "../components/projects/projects.component";

export class Project{
    id: number = 0;
    login: string = "";
    name: string = "";
    description: string = "";
    stack: string = "";

    constructor(){

    }

    parseHTTPResponse(r: GetProjectResponse){
        this.id = r.id;
        this.login = r.login;
        this.name = r.name;
        this.description = r.description;
        this.stack = r.stack;
    }

    generateCreateForm(): FormData{
        const form = new FormData();
        form.append("name", this.name);
        form.append("description", this.description);
        form.append("stack", this.stack);
        return form;
    }

    generateUpdateForm(): FormData{
        const form = new FormData();
        form.append("name", this.name);
        form.append("description", this.description);
        form.append("stack", this.stack);
        return form;
    }
}